modules/post/windows/gather/sharepoint_document_extractor.rb

```diff
@@ -0,0 +1,222 @@
+# frozen_string_literal: true
+
+# Gathers documents from a SharePoint library using the .NET API.
+# Supports HTTP and Meterpreter exfiltration with size filtering.
+class MetasploitModule < Msf::Post
+  include Msf::Post::File
+  include Msf::Post::Windows::Powershell
+
+  Rank = NormalRanking
+
+  def initialize(info = {})
+    super(update_info(info, metadata))
+    register_module_options
+  end
```

Comment:
Our initialize methods typically inline this information. While what you've written may be acceptable Ruby code, it would help us ensure consistency if we followed the same pattern as other modules here. I'm mostly worried about automated scripts that may not be owned by us that are extracting information via regexes instead of actually parsing the Ruby code.